fakebook/src/app/services/newsfeed.service.ts

```diff
@@ -29,6 +29,22 @@ export class NewsfeedService {
     );
   }
 
+  getPostById(postId: number): Promise<Post[]> {
+    const accessToken = this.oktaAuth.getAccessToken();
+    const headers = {
+      Authorization: 'Bearer ' + accessToken,
+      Accept: 'application/json',
+    };
+    let post: Post;
+    return this.http.get<Post>(
+      `${environment.baseUrls.posts}/api/posts/${postId}`,
+      { headers }
+    ).toPromise().then(p => {
+      post = p;    
+      return [post];
+    });
+  }
```

Comment:
why are you converting it to a promise and wrapping it in an array? that's silly. if you want one post it should be an observable of one post.

Reply:
Fixed according to suggestions.